jib-core/src/main/java/com/google/cloud/tools/jib/registry/ManifestPusher.java

```diff
@@ -85,8 +92,28 @@ public Void handleHttpResponseException(HttpResponseException httpResponseExcept
   }
 
   @Override
-  public Void handleResponse(Response response) {
-    return null;
+  public DescriptorDigest handleResponse(Response response)
+      throws IOException, UnexpectedImageDigestException {
+    // Checks if the image digest is as expected.
+    DescriptorDigest expectedDigest =
+        JsonTemplateMapper.toBlob(manifestTemplate)
+            .writeTo(ByteStreams.nullOutputStream())
+            .getDigest();
+
+    List<String> receivedDigests = response.getHeader(RESPONSE_DIGEST_HEADER);
+    if (receivedDigests.size() == 1) {
+      try {
+        DescriptorDigest receivedDigest = DescriptorDigest.fromDigest(receivedDigests.get(0));
+        if (expectedDigest.equals(receivedDigest)) {
+          return expectedDigest;
+        }
+
+      } catch (DigestException ex) {
+        // Invalid digest.
+      }
+    }
+
+    throw new UnexpectedImageDigestException(expectedDigest, receivedDigests);
```

Comment:
Just be cautious here, since this changes the current behavior in that Jib will start to fail for docker registries that do not return this header (probably not conforming to the spec). Do you think this will be OK? Like, what if we just show a warning and proceed with returning `expectedDigest`?

Reply:
That's a good idea, I'll make this just a warning.